static/app/components/events/interfaces/crashContent/exception/useSourceMapDebug.tsx

```diff
@@ -0,0 +1,138 @@
+import uniqBy from 'lodash/uniqBy';
+
+import type {ExceptionValue, Frame} from 'sentry/types';
+import {defined} from 'sentry/utils';
+import {QueryKey, useQuery, UseQueryOptions} from 'sentry/utils/queryClient';
+
+interface BaseSourceMapDebugError {
+  message: string;
+  type: SourceMapProcessingIssueType;
+}
+
+interface UnknownErrorDebugError extends BaseSourceMapDebugError {
+  type: SourceMapProcessingIssueType.UNKNOWN_ERROR;
+}
+interface MissingReleaseDebugError extends BaseSourceMapDebugError {
+  type: SourceMapProcessingIssueType.MISSING_RELEASE;
+}
+interface MissingUserAgentDebugError extends BaseSourceMapDebugError {
+  type: SourceMapProcessingIssueType.MISSING_USER_AGENT;
+}
+interface MissingSourcemapsDebugError extends BaseSourceMapDebugError {
+  type: SourceMapProcessingIssueType.MISSING_SOURCEMAPS;
+}
+interface UrlNotValidDebugError extends BaseSourceMapDebugError {
+  data: {absValue: string};
+  type: SourceMapProcessingIssueType.URL_NOT_VALID;
+}
+interface PartialMatchDebugError extends BaseSourceMapDebugError {
+  data: {insertPath: string; matchedSourcemapPath: string};
+  type: SourceMapProcessingIssueType.PARTIAL_MATCH;
+}
+
+export type SourceMapDebugError =
+  | UnknownErrorDebugError
+  | MissingReleaseDebugError
+  | MissingUserAgentDebugError
+  | MissingSourcemapsDebugError
+  | UrlNotValidDebugError
+  | PartialMatchDebugError;
+
+export interface SourceMapDebugResponse {
+  errors: SourceMapDebugError[];
+}
+
+export enum SourceMapProcessingIssueType {
+  UNKNOWN_ERROR = 'unknown_error',
+  MISSING_RELEASE = 'no_release_on_event',
+  MISSING_USER_AGENT = 'no_user_agent_on_release',
+  MISSING_SOURCEMAPS = 'no_sourcemaps_on_release',
+  URL_NOT_VALID = 'url_not_valid',
+  PARTIAL_MATCH = 'partial_match',
+}
+
+const sourceMapDebugQuery = ({
+  orgSlug,
+  projectSlug,
+  eventId,
+  frameIdx,
+  exceptionIdx,
+}: UseSourceMapDebugProps): QueryKey => [
+  `/projects/${orgSlug}/${projectSlug}/events/${eventId}/source-map-debug/`,
+  {
+    query: {
+      frame_idx: `${frameIdx}`,
+      exception_idx: `${exceptionIdx}`,
+    },
+  },
+];
+
+interface UseSourceMapDebugProps {
+  eventId: string;
+  exceptionIdx: number;
+  frameIdx: number;
+  orgSlug: string;
+  projectSlug: string;
+}
+
+export type StacktraceFilenameQuery = {filename: string; query: UseSourceMapDebugProps};
+
+export function useSourceMapDebug(
+  props?: UseSourceMapDebugProps,
+  options: Partial<UseQueryOptions<SourceMapDebugResponse>> = {}
+) {
+  return useQuery<SourceMapDebugResponse>(props ? sourceMapDebugQuery(props) : [''], {
+    staleTime: Infinity,
+    retry: false,
+    refetchOnWindowFocus: false,
```

Comment:
Pretty sure `refetchOnWindowFocus: false` isn't necessary with `staleTime: Infinity` since react-query will never want to refetch.

Reply:
Agree. According to some work I did last week, when Infinity is set, react query won't refetch as it will also be considered fresh. It will instead always read from the cache.

> StaleTime: The duration until a query transitions from fresh to stale. As long as the query is fresh, data will always be read from the cache only - no network request will happen! If the query is stale (which per default is: instantly), you will still get data from the cache, but a background refetch can happen under certain conditions.